src/get-started/java-binding.md

```diff
@@ -491,93 +497,318 @@ as specified in [How to Run a Service section](#how-to-run-a-service).
+  Block block = testKit.createBlockWithTransactions(validTx);
+
+  // Retrieve a snapshot of the current database state
+  Snapshot view = testkit.getSnapshot();
+  // It can be used to access the core schema, for example to check the
+  // transaction execution result:
+  Blockchain blockchain = Blockchain.newInstance(view);
+  Optional<TransactionResult> validTxResult =
+        blockchain.getTxResult(validTx.hash());
+  assertThat(validTxResult).hasValue(TransactionResult.successful());
+  // And also to verify the changes the transaction made to the service state:
+  MySchema schema = new MySchema(view);
+  // Perform assertions on the data in the service schema
+}
+```
+
+And a test that verifies that a transaction that throws an exception during its
+execution will fail:
+
+```java
+try (TestKit testKit = TestKit.forService(MyServiceModule.class)) {
+  // Construct a transaction that throws `TransactionExecutionException` during
+  // execution
+  byte errorCode = 1;
+  String errorDescription = "Test";
+  TransactionMessage errorTx =
+      constructErrorTransaction(errorCode, errorDescription);
+
+  // Commit block with this transaction
+  Block block = testKit.createBlockWithTransactions(errorTx);
+
+  // Check that transaction failed
+  Snapshot view = testKit.getSnapshot();
+  Blockchain blockchain = Blockchain.newInstance(view);
+
+  Optional<TransactionResult> errorTxResult =
+      blockchain.getTxResult(errorTx.hash());
+  TransactionResult expectedTransactionResult =
+      TransactionResult.error(errorCode, errorDescription);
+  assertThat(errorTxResult).hasValue(expectedTransactionResult);
+}
+```
+
+The TestKit also allows creating blocks that contain all current [in-pool][in-pool]
+transactions:
+
+```java
+try (TestKit testKit = TestKit.forService(MyServiceModule.class)) {
+  // Put the transaction into the TestKit transaction pool
+  MyService service = testKit.getService(MyService.SERVICE_ID, MyService.class);
+
+  TransactionMessage message = constructTransactionMessage();
+  RawTransaction rawTransaction = RawTransaction.fromMessage(message);
+  service.getNode().submitTransaction(rawTransaction);
 
-### Read Requests
+  // This block will contain the transaction submitted above
+  Block block = testKit.createBlock();
+  // Check the resulting block or blockchain state
+}
+```
 
-To test read requests for service data, you can use a fake that implements
-`Node` interface and uses `MemoryDb` to create `Snapshot`:
-[`NodeFake`][nodefake].
-The `MemoryDb` contents can be filled in by executing `MemoryDb#merge(Fork)`
-operation as in the section above.
+The TestKit provides [`getTransactionPool()`][testkit-get-pool] and
+[`findTransactionsInPool(Predicate<TransactionMessage> predicate)`][testkit-find-in-pool]
+methods to inspect the transaction pool. It is useful when there is a need to verify
+transactions that the service instance submitted itself (e.g., in `afterCommit`
+method), as those are put into the pool.
```

Comment:
These methods are useful when there is a need to verify transactions that the service instance submitted itself (e.g., in `afterCommit` method) into the transaction pool.